src/openms/include/OpenMS/ANALYSIS/ID/PeptideIndexing.h

```diff
@@ -744,220 +740,39 @@ namespace OpenMS
-         const std::pair<int, int>& prefix_suffix_counts = pair.second;
-         double freq_prefix = static_cast<double>(prefix_suffix_counts.first) / static_cast<double>(all_prefix_occur);
-         double freq_prefix_in_proteins = static_cast<double>(prefix_suffix_counts.first) / static_cast<double>(all_proteins_count);
-
-         if (freq_prefix >= 0.8 && freq_prefix_in_proteins >= 0.4)
-         {
-           prefix_ = true;
-           decoy_string_ = decoy_case_sensitive[case_insensitive_decoy_string];
-
-           if (prefix_suffix_counts.first != all_prefix_occur)
-           {
-             OPENMS_LOG_WARN << "More than one decoy prefix observed!" << std::endl;
-             OPENMS_LOG_WARN << "Using most frequent decoy prefix (" << (int) (freq_prefix * 100) <<"%)" << std::endl;
-           }
-
-           return true;
-         }
-       }
-
-       // Decoy suffix occurred at least 80% of all suffixes + observed in at least 40% of all proteins -> set it as suffix decoy
-       for (const auto& pair : decoy_count)
-       {
-         const std::string& case_insensitive_decoy_string = pair.first;
-         const std::pair<int, int>& prefix_suffix_counts = pair.second;
-         double freq_suffix = static_cast<double>(prefix_suffix_counts.second) / static_cast<double>(all_suffix_occur);
-         double freq_suffix_in_proteins = static_cast<double>(prefix_suffix_counts.second) / static_cast<double>(all_proteins_count);
-
-         if (freq_suffix >= 0.8 && freq_suffix_in_proteins >= 0.4)
-         {
-           prefix_ = false;
-           decoy_string_ = decoy_case_sensitive[case_insensitive_decoy_string];
-
-           if (prefix_suffix_counts.second != all_suffix_occur)
-           {
-             OPENMS_LOG_WARN << "More than one decoy suffix observed!" << std::endl;
-             OPENMS_LOG_WARN << "Using most frequent decoy suffix (" << (int) (freq_suffix * 100) <<"%)" << std::endl;
-           }
-
-           return true;
-         }
-       }
-
-       OPENMS_LOG_ERROR << "Unable to determine decoy string and its position. Please provide a decoy string and its position as parameters." << std::endl;
-       return false;
-     }
-
-
-     struct PeptideProteinMatchInformation
-    {
-      /// index of the protein the peptide is contained in
-      OpenMS::Size protein_index;
-
-      /// the position of the peptide in the protein
-      OpenMS::Int position;
-
-      /// the amino acid after the peptide in the protein
-      char AABefore;
 
-      /// the amino acid before the peptide in the protein
-      char AAAfter;
+    struct PeptideProteinMatchInformation
+    {
+      OpenMS::Size protein_index; //< index of the protein the peptide is contained in
+      OpenMS::Int position; //< the position of the peptide in the protein
+      char AABefore; //< the amino acid after the peptide in the protein
+      char AAAfter; //< the amino acid before the peptide in the protein
 
+      const auto tie() const
```

Comment:
smells like > x11

Reply:
https://en.cppreference.com/w/cpp/utility/tuple/tie -- since C++11

Reply:
But also Auto Return value?

Reply:
true ... I dont think that works yet

Reply:
it works on MSVC2015.. and I thought I give it a try. Well...